### PR TITLE
Housekeeping: Remove useless code, tidy up code, remove inconsistent use of options

### PR DESCRIPTION
**Pull request by icebound777**
* simpler code for random_shop_prices
* move optionset.py into models directory; split the different optionsets into multiple files
* refactor away the inconsistent use of accessing options (raw versus '["value"]')
* fix some formatting
* remove useless comments
* remove useless code from logic._find_new_nodes_and_edges

Overall cleaner code, and for some reason a performance gain of 7-8%, which is nice

Code has been tested by generating and comparing several seeds on this branch's top commit of d0475a2e against the same seeds on the dev branch 20118d15 using the extreme shuffle preset, after manipulating the seed hash items to be same the resulting md5 checksum matched.

### CODE REVIEW

File: rando_modules/logic.py [hunk 1]
```diff
@@ -204,41 +199,18 @@ def _find_new_nodes_and_edges(
...
-        #logging.debug("non_traversable_edges_cpy after %s", non_traversable_edges_cpy)
-        #logging.debug("non_traversable_edges after %s", non_traversable_edges)
-
-        # Check if an item node is reachable which already has an item placed.
-        # Since nodes are usually removed from this dict the moment an item is
-        # placed there, this can only be true for node which's item is not being
-        # randomized (caused by turned off shop shuffle, coin shuffle or
-        # hidden panel shuffle).
-        # If this is the case add item to inventory and remove node
-        for node_id, item_node in reachable_item_nodes.items():
```
Comment (Pronyo-Chan) [inline rando_modules/logic.py:216]: What was that loop not needed at all for the longest time..?

File: rando_modules/modify_entrances.py [hunk 2]
```diff
@@ -295,27 +296,26 @@ def get_bowsercastle_bossrush(world_graph: dict):
...
-    world_graph, hos_entrance_modifications = adjust(
-        world_graph,
-        new_edges=edges_hos_add,
-        edges_to_remove=edges_hos_remove
-    )
-    world_graph, kpa_entrance_modifications = adjust(
+    all_new_edges.extend(edges_hos_add)
+    all_new_edges.extend(edges_kpa_bossrush_add)
+    all_edges_to_remove.extend(edges_hos_remove)
+    all_edges_to_remove.extend(edges_kpa_bossrus_remove)
```
Comment (Pronyo-Chan) [inline rando_modules/modify_entrances.py:306]: typo in "bossrush"

File: rando_modules/modify_entrances.py [hunk 3]
```diff
@@ -295,27 +296,26 @@ def get_bowsercastle_bossrush(world_graph: dict):
...
+    all_edges_to_remove = []
+    entrance_modifications = []
 
-    world_graph, hos_entrance_modifications = adjust(
-        world_graph,
-        new_edges=edges_hos_add,
-        edges_to_remove=edges_hos_remove
-    )
-    world_graph, kpa_entrance_modifications = adjust(
+    all_new_edges.extend(edges_hos_add)
```
Comment (Pronyo-Chan) [inline rando_modules/modify_entrances.py:303]: Do we need to deepcopy the edges? Having PTSD from the star hunt saga where not doing it caused issues. Not sure if it applies here.

File: rando_modules/logic.py (hunk 1 above)
Comment (icebound777) [reply rando_modules/logic.py:216]: This code was used in the old forward fill algorithm. The function for writing the spoiler log used this function as well, but didn't use several of the local variables, so the loop wasn't actually needed.
During the switch from forward fill to assumed fill the actual reason for this code being here was dropped, but this function was never refactored to remove the unused code.

File: rando_modules/modify_entrances.py (hunk 3 above)
Comment (icebound777) [reply rando_modules/modify_entrances.py:303]: We do not need to deepcopy here. That is only needed if we either manipulate an edge afterwards (like in get_starhunt) or use the same edge multiple times (like in _set_rowf_out_of_logic etc.), both of which we do not do here.

**Pull request merged**